pkg/identity/numericidentity.go

```diff
@@ -55,8 +58,139 @@ const (
+			return i.identity
+		}
+	}
+
+	return nil
+}
+
+func (w wellKnownIdentities) lookupByNumericIdentity(identity NumericIdentity) *Identity {
+	wki, ok := w[identity]
+	if !ok {
+		return nil
+	}
+	return wki.identity
+}
+
+// initWellKnownIdentities establishes all well-known identities
+func initWellKnownIdentities() {
+	// etcd-operator labels
+	//   k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa
+	//   k8s:io.kubernetes.pod.namespace=kube-system
+	//   k8s:io.cilium/app=etcd-operator
+	//   k8s:io.cilium.k8s.policy.cluster=default
+	wellKnown.add(ReservedETCDOperator, []string{
+		"k8s:io.cilium/app=etcd-operator",
+		fmt.Sprintf("k8s:%s=kube-system", api.PodNamespaceLabel),
+		fmt.Sprintf("k8s:%s=cilium-etcd-sa", api.PolicyLabelServiceAccount),
+		fmt.Sprintf("k8s:%s=%s", api.PolicyLabelCluster, option.Config.ClusterName),
+	})
+
+	// cilium-etcd labels
+	//   k8s:app=etcd
+	//   k8s:io.cilium/app=etcd-operator
+	//   k8s:etcd_cluster=cilium-etcd
+	//   k8s:io.cilium.k8s.policy.serviceaccount=default
+	//   k8s:io.kubernetes.pod.namespace=kube-system
+	//   k8s:io.cilium.k8s.policy.cluster=default
+	// these 2 labels are ignored by cilium-agent as they can change over time
+	//   container:annotation.etcd.version=3.3.9
+	//   k8s:etcd_node=cilium-etcd-6snk6vsjcm
+	wellKnown.add(ReservedCiliumKVStore, []string{
+		"k8s:app=etcd",
+		"k8s:etcd_cluster=cilium-etcd",
+		"k8s:io.cilium/app=etcd-operator",
+		fmt.Sprintf("k8s:%s=kube-system", api.PodNamespaceLabel),
+		fmt.Sprintf("k8s:%s=default", api.PolicyLabelServiceAccount),
+		fmt.Sprintf("k8s:%s=%s", api.PolicyLabelCluster, option.Config.ClusterName),
+	})
+
+	// kube-dns labels
+	//   k8s:io.cilium.k8s.policy.serviceaccount=kube-dns
+	//   k8s:io.kubernetes.pod.namespace=kube-system
+	//   k8s:k8s-app=kube-dns
+	//   k8s:io.cilium.k8s.policy.cluster=default
+	wellKnown.add(ReservedKubeDNS, []string{
+		"k8s:k8s-app=kube-dns",
+		fmt.Sprintf("k8s:%s=kube-system", api.PodNamespaceLabel),
+		fmt.Sprintf("k8s:%s=kube-dns", api.PolicyLabelServiceAccount),
+		fmt.Sprintf("k8s:%s=%s", api.PolicyLabelCluster, option.Config.ClusterName),
+	})
+
+	// kube-dns EKS labels
+	//   k8s:io.cilium.k8s.policy.serviceaccount=kube-dns
+	//   k8s:io.kubernetes.pod.namespace=kube-system
+	//   k8s:k8s-app=kube-dns
+	//   k8s:io.cilium.k8s.policy.cluster=default
+	//   k8s:eks.amazonaws.com/component=kube-dns
+	wellKnown.add(ReservedEKSKubeDNS, []string{
+		"k8s:k8s-app=kube-dns",
+		"k8s:eks.amazonaws.com/component=kube-dns",
+		fmt.Sprintf("k8s:%s=kube-system", api.PodNamespaceLabel),
+		fmt.Sprintf("k8s:%s=kube-dns", api.PolicyLabelServiceAccount),
+		fmt.Sprintf("k8s:%s=%s", api.PolicyLabelCluster, option.Config.ClusterName),
+	})
+
+	// CoreDNS labels
+	//   k8s:io.cilium.k8s.policy.serviceaccount=coredns
+	//   k8s:io.kubernetes.pod.namespace=kube-system
+	//   k8s:k8s-app=kube-dns
+	//   k8s:io.cilium.k8s.policy.cluster=default
+	wellKnown.add(ReservedCoreDNS, []string{
+		"k8s:k8s-app=kube-dns",
```

Comment:
You repeated this line two times, maybe use a cons here?

Reply:
I don't think it's a good idea to use constants here because these things may change and having a single constant for things like the kube-dns label can lead to blindly adjusting that constant that will then affect well-known identities. Once merged, we can never change these well-known identities anymore.